src/control/lib/hardware/fabric.go

```diff
@@ -812,3 +858,80 @@ func (s *FabricScanner) CacheTopology(t *Topology) error {
 	s.topo = t
 	return nil
 }
+
+// FabricReadyChecker is an interface for a type that can be used to check whether a fabric
+// interface is ready.
+type FabricReadyChecker interface {
+	CheckFabricReady(string) error
+}
+
+// WaitFabricReadyParams defines the parameters for a WaitFabricReady call.
+type WaitFabricReadyParams struct {
+	Checker        FabricReadyChecker
+	FabricIfaces   []string      // Fabric interfaces that must become ready
+	IterationSleep time.Duration // Time between iterations if some interfaces aren't ready
```

Comment:
do we want to have an exponential backoff rather than just a fixed duration and do we want a limit/cap after which we just return an error or does another high-level timeout cover this?

Reply:
From my perspective I don't think an exponential backoff makes sense here. Usually those are employed to avoid a "thundering herd" problem where many clients are all attempting to reach a service that is starting up or restarting. In this case we want the server to be up and responsive as quickly as possible, and I don't think that checking the interface status will have an appreciable impact on the system.

I'm not sure that we want an arbitrary timeout, either, as I think it's difficult to choose a value that is likely to work in all situations. There is a logged error on every iteration of the loop where the interface is not ready, so the admin should be able to understand what the server is doing.

Reply:
Yeah, my inclination was to let it roll forever waiting (maybe the admin notices and runs to the lab to plug in the IB cable while DAOS continues spinning?). We could inject a timeout via the Context, if we find one is needed.